content/blog/blog-2020-07-01-kudo-dependencies.md

```diff
@@ -0,0 +1,93 @@
+
+We've been thinking about the best way of introducing dependencies to KUDO for a while now. A few months ago we created [KEP-29](https://github.com/kudobuilder/kudo/blob/main/keps/0029-operator-dependencies.md) that aimed at solving at least part of the problems. Dependencies can be a complex topic. KEP-29 is not trying to boil the dependency ocean but rather limits itself to installation dependencies only, i.e. your operator instance and all its dependencies (including transitive ones) will be installed and/or removed as one unit.
+
+<!-- more -->
+
+## KudoOperator Task
+
+KUDO operators already have a mechanism to deal with installation dependencies called [plans, phases, and steps](plans.md) with serial or parallel execution strategy. This mechanism is already powerful enough to express any dependency hierarchy including transitive dependencies. As of KUDO 0.15.x, KUDO supports a `KudoOperator` task which allows you to specify a dependency on other operators. At the very least it requires the name of the operator to be installed. You can also provide an operator version (defaults to the latest one).
+
+Here a simple example of a task specifying a dependency on the community Zookeeper operator version 0.3.0 (which will install [zookeeper 3.4.14](https://github.com/kudobuilder/operators/blob/master/repository/zookeeper/README.md))
+
+```yaml
+tasks:
+  - name: deploy-zookeeper
+    kind: KudoOperator
+    spec:
+      package: zookeeper
+      operatorVersion: 0.3.0
+```
+
+This task can be used as part of e.g. `deploy` plan same as any other:
+
+```yaml
+plans:
+  deploy:
+    strategy: serial
+    phases:
+      - name: prereqs
+        strategy: parallel
+        steps:
+          - name: first
+            tasks:
+              - deploy-zookeeper
+      - name: main
+        strategy: parallel
+        steps:
+          - name: second
+            tasks:
+              - deploy-main
+```
+
+As with other tasks, KUDO will make sure this task is healthy before moving to the next one. _"Healthy"_, in the case of the operator means that its `deploy` plan has finished successfully. Note, that any transitive dependencies of the zookeeper operator itself will also be resolved and installed in the right order. `KudoOperator` closely mimics the `kubectl kudo install` command semantics allowing to additionally specify the `appVersion` (defaults to most recent one) and the `instanceName` (will be generated by KUDO by default).
```

Comment:
"allowing to additionally specify the " is hard to parse IMO... "allowing for the configuration of "?

Reply:
I agree - Maybe split this sentence?

KudoOperator closely mimics the kubectl kudo install comand semantics. It allows to specify the appVersion ...